Objects/object.c

```diff
@@ -300,12 +297,6 @@ PyObject_Print(PyObject *op, FILE *fp, int flags)
                     Py_DECREF(t);
                 }
             }
-            else {
-                PyErr_Format(PyExc_TypeError,
-                             "str() or repr() returned '%.100s'",
-                             Py_TYPE(s)->tp_name);
-                ret = -1;
-            }
             Py_XDECREF(s);
```

Comment:
As a micro-optimization, this can now be moved inside the else block and turned into `Py_DECREF`.